### PR TITLE
support http2 tracer

### PR DESCRIPTION
**Pull request by OrezzerO**
### Issues associated with this PR

support http2 tracer

### CODE REVIEW

**Comment by OrezzerO** [comment]
We can implement our own  Tracer:


```go
func init() {
	trace.RegisterTracerBuilder("SOFATracer", protocol.HTTP2, NewTracer)
}

type HttpTracer2 struct{}

func NewTracer(config map[string]interface{}) (types.Tracer, error) {
	return &HttpTracer2{}, nil
}
``` 
 Then, we can see http2 trace log  at. `~/logs/tracelog/rpc-client-digest.log` and `~/logs/tracelog/rpc-server-digest.log`
image

**Review by nejisama — APPROVED** [review]

File: pkg/stream/http2/stream.go [hunk 1]
```diff
@@ -413,8 +416,17 @@ func (conn *serverStreamConnection) onNewStreamDetect(ctx context.Context, h2s *
 	conn.streams[stream.id] = stream
 	conn.mutex.Unlock()
 
-	stream.receiver = conn.serverCallbacks.NewStreamDetect(stream.ctx, stream, nil)
+	var span types.Span
```
Comment (nejisama) [inline pkg/stream/http2/stream.go:419]: 这个已经被重构成api.Span了 调整一下

Comment (taoyuanyuan) [inline pkg/stream/http2/stream.go:419]: merge master, `types.Span` have moved to `api.Span`

Comment (OrezzerO) [reply pkg/stream/http2/stream.go]: done

**Review by nejisama — APPROVED** [review]

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**